llvm/lib/Support/RISCVISAInfo.cpp

```diff
@@ -466,35 +466,35 @@ bool RISCVISAInfo::compareExtension(const std::string &LHS,
   return LHS < RHS;
 }
 
-void RISCVISAInfo::toFeatures(
-    std::vector<StringRef> &Features,
-    llvm::function_ref<StringRef(const Twine &)> StrAlloc,
-    bool AddAllExtensions) const {
-  for (auto const &Ext : Exts) {
-    StringRef ExtName = Ext.first;
-
-    if (ExtName == "i")
+std::vector<std::string> RISCVISAInfo::toFeatures(bool AddAllExtensions,
+                                                  bool IgnoreUnknown) const {
+  std::vector<std::string> Features;
+  for (const auto &[ExtName, _] : Exts) {
+    if (ExtName == "i") // i is not recognized in clang -cc1
```

Comment:
Could you please add a comment explaining that this is the RVI i extension with a link to the spec?  I fear that non-RISCV developers might be confused by the random check for `i`.

Reply:
This was just copied from the previous function, but would non-RISCV developers need to read RISCVISAInfo.cpp? (Still happy to add a link to the spec if you would prefer)

Reply:
Yeah, sometimes changes do require looking at other architectures.